pandas/io/parsers.py

```diff
@@ -2273,11 +2273,15 @@ def __init__(self, f, **kwds):
         # Get columns in two steps: infer from data, then
         # infer column indices from self.usecols if it is specified.
         self._col_indices = None
-        (
-            self.columns,
-            self.num_original_columns,
-            self.unnamed_cols,
-        ) = self._infer_columns()
+        try:
+            (
+                self.columns,
+                self.num_original_columns,
+                self.unnamed_cols,
+            ) = self._infer_columns()
+        except EmptyDataError:
+            self.close()
+            raise
```

Comment:
this should be

```
try: 
......
except EmptyDataError as err:
      self.close()
      raise from err
```

Reply:
also I think ok to catch TypeError, ValueError, EmptyDataError here (all things raised by infer_columns)

Reply:
The raise from clause only works in the case when you change the exception type, not in this case, so if you do `raise MyNewException from err`
`EmptyDataError` is derived from `ValueError` so I changed it to catch `ValueError` and `TypeError`